pkg/alertmanager/api.go

```diff
@@ -184,3 +185,42 @@ func validateUserConfig(logger log.Logger, cfg alertspb.AlertConfigDesc) error {
 
 	return nil
 }
+
+func (am *MultitenantAlertmanager) ListUserConfig(w http.ResponseWriter, r *http.Request) {
+	logger := util_log.WithContext(r.Context(), am.logger)
+	userIDs, err := am.store.ListAllUsers(r.Context())
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to list users of alertmanager")
+		http.Error(w, fmt.Sprintf("%s: %s", errListAllUser, err.Error()), http.StatusInternalServerError)
+		return
+	}
+	userConfigMap := make(map[string]*UserConfig, len(userIDs))
+	for _, userID := range userIDs {
+		cfg, err := am.store.GetAlertConfig(r.Context(), userID)
```

Comment:
New `GetAlertConfigs` could be used to efficiently load configs for all users at once (or perhaps smaller batches). 

I'm bit afraid what happens when someone calls this on large alertmanager installation with thousands of users -- this will be slow and can take a lot of memory, possibly crashing AM. It would be best to stream output instead of building it in memory first. WDYT?

Reply:
Update to use GetAlertConfigs.